File: client/src/models/schedule.ts
export interface ISchedule {
  id: string;
  startTime: string;
  endTime: string;
  status: number;
  hours: number;
  userId: string;
  groupId: string;
  siteId: string;
  type: number;
}

export interface IAddSchedule {
  startTime: string;
  endTime: string;
  status: number;
  userId: string;
  groupId: string;
  siteId: string;
  type: number;
}
export interface IAddScheduleTimeOff {
  dates: string[];
  status: number;
  userId: string;
  groupId: string;
  siteId: string;
  type: number;
}

export interface IEditSchedule {
  type: number;
  id: string;
  startTime: string;
  endTime: string;
  status: number;
  userId: string;
  groupId: string;
  siteId: string;
}

File: API/Models/Schedule.cs
namespace API.Models
{
    public class Schedule : BaseEntity
    {
        public string Id { get; set; }
        private DateTime _startTime;
        public DateTime StartTime
        {
            get => _startTime;
            set => _startTime = DateTime.SpecifyKind(value, DateTimeKind.Utc);
        }
        private DateTime _endTime;
        public DateTime EndTime
        {
            get => _endTime;
            set => _endTime = DateTime.SpecifyKind(value, DateTimeKind.Utc);
        }
        public string SiteId { get; set; }
        public Site Site { get; set; }
        public string UserId { get; set; }
        public User User { get; set; }
        public string GroupId { get; set; }
        public Group Group { get; set; }
        public int Type { get; set; }
        public int Status { get; set; }
    }
}